R/features.R

```diff
@@ -112,34 +111,33 @@ feature_not_exact <- function(m, n_combinations = 200, weight_zero_m = 10^6, red
 
   # Get number of occurences and duplicated rows-------
   r <- helper_feature(m, feature_sample)
-  X[, no := r[["no"]]]
   X[, is_duplicate := r[["is_duplicate"]]]
-  X[, ID := .I]
+
+  # When we sample combinations the Shapley weight is equal
+  # to the frequency of the given combination
+  X[, shapley_weight := r[["sample_frequence"]]]
 
   # Populate table and remove duplicated rows -------
   X[, features := feature_sample]
-  if (reduce_dim && any(X[["is_duplicate"]])) {
+  if (any(X[["is_duplicate"]])) {
     X <- X[is_duplicate == FALSE]
-    X[, no := 1]
   }
   X[, is_duplicate := NULL]
-  nms <- c("ID", "nfeatures", "features", "no")
-  data.table::setcolorder(X, nms)
 
   # Add shapley weight and number of combinations
-  X[, shapley_weight := weight_zero_m]
+  X[c(1, .N), shapley_weight := weight_zero_m]
   X[, N := 1]
   X[between(nfeatures, 1, m - 1), ind := TRUE]
-  X[ind == TRUE, shapley_weight := p[nfeatures]]
+  X[ind == TRUE, p := p[nfeatures]]
```

Comment:
Is this used to get the ranking in decent order, or something else?

Reply:
I want to keep the sampling probabilities, so that we can make sure that they are correct. This is done by unit tests in `tests/testthat/test-features.R`.